bin/i18n/sync-in.rb

```diff
@@ -440,7 +441,41 @@ def localize_course_offerings
   CourseOffering.all.sort.each do |co|
     hash[co.key] = co.display_name
   end
-  File.open(File.join(I18N_SOURCE_DIR, "dashboard/course_offerings.json"), "w+") do |f|
+  write_dashboard_json('course_offerings', hash)
+end
+
+# Encapsulation of the sync-in process for our `Script`
+# model's attributes. This is meant to be built on top of
+# for any future attributes we end up needing to translate.
+def localize_scripts
+  puts "Preparing scripts"
+
+  scripts_hash = Hash.new
+  Script.all.each do |script|
```

Comment:
Maybe this should just be localize_announcements? We already have a component which localizes scrips: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/lib/services/i18n/curriculum_sync_utils/sync_in.rb#L18

I tried taking a quick look and these announcements are different than existing Script attributes since the announcements don't have an Active Record model. I'm not sure if this announcement sync-in should be done in `sync-in.rb` or `sync_in.rb`.

I'm hoping you have time to dive a little deeper into it and let me know where you think it should live.

Reply:
Looking at `CurriculumSyncUtils` it has the documentation:
```
    # This module contains all i18n sync logic specific to the content
    # imported into Code Studio from CurriculumBuilder; Lessons, Activities,
    # Resources, Vocabularies, Objectives, etc.
```
From your PR, it looks like these announcements are created on Levelbuilder, so I think we do want `CurriculumSyncUtils` to handle this.

Reply:
nice find Dayne. I would add that this comment is a bit backwards-looking, since CurriculumBuilder will soon be a thing of the past. I would say it is for translating all curriculum content including unit groups, scripts, lessons, and associated objects (including activities, resources, vocab, objectives), but not including levels.